packages/web/components/bridge/immersive/amount-and-review-screen.tsx

```diff
@@ -122,16 +130,40 @@ export const AmountAndConfirmationScreen = observer(
               fiatAmount={fiatAmount}
               setFiatAmount={setFiatAmount}
               quote={quote}
+              onConfirm={() => setCurrentScreen(ImmersiveBridgeScreens.Review)}
             />
           )}
         </Screen>
         <Screen screenName={ImmersiveBridgeScreens.Review}>
           {({ goBack }) => (
-            <div>
-              <h6>Step 3: Review</h6>
-              <Button onClick={goBack}>Back</Button>
-              <Button onClick={onClose}>Close</Button>
-            </div>
+            <>
+              {fromChain &&
+                toChain &&
+                fromAddress &&
+                toAddress &&
+                fromWalletIcon &&
+                toWalletIcon &&
+                sourceAsset &&
```

Comment:
What if we add this conditional within the review screen, similar to what I did in the amount screen? However, instead of displaying a loading indicator, we throw an error because it shouldn't be possible to proceed without this info

Reply:
I think the component code & props are much cleaner and less error prone within the review component if we validate that those are present before passing them to the component. Perhaps Instead of showing nothing we can display a text error message in the UI?

Reply:
Yes, we should display an error message. I believe we should throw the error in the component so we can catch it with an ErrorBoundary and share the fallback between components